EXPORT_HELPERS.md

```diff
@@ -0,0 +1,215 @@
+
+Goal of this proposal is to produce consistent and conformant formatted exports of the VSS via
+the VSS-TOOLS or other potential 3rd-party implementations, resulting in a high degree of stability and
+interoperability between implementations using a particular format (such as IDL).
+
+This consistency and stability can ease the creation of interoperable examples and reference implementations for a given format.  
+This draft focuses on IDL, but is intended to also maximize interoperability within other export formats.
+
+To enable this capability, the following changes are proposed:
+
+- Create a document to describe the recommended formatting and structure of the resulting exported (IDL or other) files.
+- Modify the exporters in VSS-TOOLS to conform to the above document.
+- Utilize the `arraysize` and `stringsizemax` to set the dimensions of the exported array/sequence/string types.
+- Introduce 3 new keywords to the VSS: `memberof`, `membergroup`, `isogroup`, used as:
+    - To have a `branch` entity create a namespace in the export (named for itself), add:
+        - `memberof: namespace`
+    - To create named group containers in the export for elements, add to the branch entry:
+        - `membergroup <list of names>` to create containers that may also be referenced in the parent of this branch.
+        - `isogroup <list of names>` to create containers that will not be referenced in the parent of this branch.
+    - To assign `membergroup` struct names to be referenced in a parent struct:
+        - `memberof: <list of parent struct names to insert reference into>`
+    - To assign VSS elements to specific named group containers in the export, add to element entry:
+        - `memberof: <list of group containers>`
+
+## Effect on IDL Export
+Implementing the above keywords into the VSS -- to enforce the guidance of the above reference document -- 
+should have no effect on existing exporters in VSS-TOOLS, however a modified IDL exporter could use this new
+information to consistently produce IDL that conforms to the above document.
+
+Examples of this export guidance on IDL includes:  
+
+### Namespaces
+In IDL, namespaces are expressed with a `module { ... }` container around the affected definitions.
+A sensible approach to namespacing in VSS would be to use containers for the major sections of
+the specification, such as: `ADAS`, `Body`, `Cabin`, `Chassis`, `OBD`, and `Powertrain`, all residing
+within a top-level container for `Vehicle`.   The resulting IDL would look like:
+```
+  module Vehicle {
+    module Cabin {
+      (struct and enum definitions for Cabin descendants)
+    };
+    module Chassis {
+      (struct and enum definitions for Chassis descendants)
+    };
+  };
+```
+Note that only the `branch` entries that have this keyword will receive a namespace for their descendent types.
+All `branch` entries beneath them will be implemented as `struct`.
+
+### Creating structs from VSS elements

```

Comment:
Generally, it makes sense to group signals into structs, but how does this "tagging" approach later merges/matches with "real structs" as we recently defined

Reply:
However, I agree that if both PR's are implemented, it could represent 2 different ways of specifying a struct:
269 enforces using a struct (new signals only)
576 suggests a struct and is optional (all signals)